### DISCUSSION TITLE
adding URL to column breaks split layout

### DISCUSSION THREAD

**Discussion by jdion84** [filamentphp/filament#9879] [Help]
when adding a `->url()` to a text column, it messes up a split table layout completely

before adding `->url()`:

![](https://i.imgur.com/KjY12Rj.png)

after adding `->url()`:

![](https://i.imgur.com/xFmLWSy.png)

here is an example of the code i am using for this table:

```php
        return $table
            ->columns([
                Split::make([
                    TextColumn::make('name')
                        ->searchable()
                        ->sortable(),

                    Stack::make([
                        TextColumn::make('email')
                            ->url(function ($state) {
                                return "mailto:$state";
                            })
                            ->searchable()
                            ->sortable(),

                        TextColumn::make('phone')
                            ->searchable()
                            ->sortable(),
                    ]),

                    TextColumn::make('type')
                        ->sortable(),
                ]),
            ])->from('md')
```

i prefer using the `Split` for all tables because its amazing on mobile.

it appears this may be caused by the `w-full` class applied to URL's inside table columns.

**Comment by leandrocfe** [comment]
Not sure if it works
```php
->formatStateUsing(fn ($state) => new HtmlString('<a href="xx">'.$state.'</a>'))
```

**Reply by jdion84** [reply]
its the `a` tag in general. for example, if i use a `span` instead, its fine

its some styling to do with the `a` tag i can't figure out after reading through the dev tools repeatedly
